src/lib.rs

```diff
@@ -90,13 +91,18 @@ pub struct ZWaveManager {
 }
 
 impl ZWaveManager {
-    fn new(manager: manager::Manager) -> ZWaveManager {
-        ZWaveManager {
+    fn new(manager: manager::Manager) -> (Self, mpsc::Receiver<ZWaveNotification>) {
+        let (tx, rx) = mpsc::channel();
+
+        let manager = ZWaveManager {
             watcher: ZWaveWatcher {
                 state: Arc::new(Mutex::new(State::new())),
+                sender: Arc::new(Mutex::new(tx))
```

Comment:
yeah I'll add one :)
basically I need a Arc because I need a Mutex... (I think ?) and we clone the structure for each watcher we add. But maybe we'd want to implement Clone differently: cloning the tx and creating another mutex on the clone. And then we can get rid of the Arc. This can be done later though, this shoudln't be a bottleneck, but I add a TODO comment here.